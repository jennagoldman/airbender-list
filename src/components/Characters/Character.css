.characterName {
  font-family: 'Lemonada', sans-serif;
}

li div {
  background: #fff;
  width: 300px;
  height: 300px;
  padding: 25px 10px;
  border-radius: 15px;
  box-shadow: 10px 10px #000;
}

li div p {
  margin: 10px;
  font-size: 24px;
}

li a {
  color: coral;
  text-shadow: 1px 1px #000;
}

li div img {
  width: 250px;
  height: 250px;
  border-radius: 10px;
}