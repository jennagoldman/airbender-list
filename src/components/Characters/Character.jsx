import React from 'react';
import PropTypes from 'prop-types';
import styles from './Character.css';

const Character = ({ name, imageUrl }) => (
  <div className={styles.characterDiv}>
    <p className={styles.characterName}>{name}</p>
    <img src={imageUrl} alt={name} />
  </div>
);

Character.propTypes = {
  name: PropTypes.string.isRequired,
  imageUrl: PropTypes.string,
};


export default Character;
