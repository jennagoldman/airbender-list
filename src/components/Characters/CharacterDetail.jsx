import React from 'react';
import PropTypes from 'prop-types';
import styles from './CharacterDetail.css';

const CharacterDetail = ({ name, affiliation, allies, enemies, photoUrl }) => (
  <div className={styles.characterDetail}>
    <p className={styles.characterName}>{name}</p>
    <img src={photoUrl} alt={name} />
    <p><i className="fas fa-users"></i> {affiliation} </p>
    <p>
      <span><i className="fas fa-hands-helping"></i> {allies} | </span>
      <span><i className="fas fa-fist-raised"></i> {enemies}</span>
    </p>
  </div>
);

CharacterDetail.propTypes = {
  name: PropTypes.string.isRequired,
  photoUrl: PropTypes.string,
  allies: PropTypes.array,
  enemies: PropTypes.array,
  affiliation: PropTypes.string,
};


export default CharacterDetail;
