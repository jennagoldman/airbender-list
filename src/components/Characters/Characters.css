ul {
  margin-block-start: 0;
  margin-block-end: 0;
  padding-inline-start: 0;
}

li {
  display: inline-block;
  margin: 20px;
  text-align: center;
  vertical-align: middle
}