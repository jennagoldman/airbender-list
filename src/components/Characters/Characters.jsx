import React from 'react';
import Character from './Character.jsx';

const Characters = ({ characters }) => {
  const characterItems = characters.map(character => (
    <li key={character.id}>
      <Character {...character} />
    </li>
  ));

  return (
    <ul>
      {characterItems}
    </ul>
  );
};

Characters.propTypes = {
  characters: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    allies: PropTypes.string.isRequired,
    enemies: PropTypes.string.isRequired,
    affiliation: PropTypes.string.isRequired
  }))
};


export default Characters;
