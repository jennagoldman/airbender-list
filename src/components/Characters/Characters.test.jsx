import React from 'react';
import { shallow } from 'enzyme';
import Characters from './Characters.jsx';

describe('Characters', () => {
  it('matches a snapshot', () => {
    const characters = [{
      id: '1',
      name: 'Momo',
      imageUrl: 'https://vignette.wikia.nocookie.net/avatar/images/4/43/Inquisitive_Momo.png/revision/latest?cb=20081225191217'
    }];
    const wrapper = shallow(<Characters characters={characters} />);
    expect(wrapper).toMatchSnapshot();
  });
});
