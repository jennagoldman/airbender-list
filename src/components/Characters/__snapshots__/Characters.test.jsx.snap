// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Characters matches a snapshot 1`] = `
<ul
  className="characterList"
>
  <li
    className="characterItem"
    key="1"
  >
    <Link
      to="/character/1"
    >
      <Character
        id="1"
        imageUrl="https://vignette.wikia.nocookie.net/avatar/images/4/43/Inquisitive_Momo.png/revision/latest?cb=20081225191217"
        name="Momo"
      />
    </Link>
  </li>
</ul>
`;
